### DISCUSSION TITLE
If the file is too large, neovim simply freezes and does not work. Is there any way to solve this problem?

### DISCUSSION THREAD

**Discussion by gengdz** [LazyVim/LazyVim#470] [Q&A]
The file size is 4M

**Comment by amaanq** [comment]
Likely from tree-sitter injections, is it a c/c++ file?

**Reply by gengdz** [reply]
You're right, it's the tree-sitter that's causing the problem.

It can be any type of file

**Comment by hakonharnes** [comment]
I'm also having this issue. And yes @amaanq, it is a c file in my case. Works just fine in vanilla vim.

**Comment by amaanq** [comment]
comment out nvim-treesitter/queries/{c,cpp}/injections.scm

**Reply by hakonharnes** [reply]
It's certainly better, but still a bit laggy. Completely disabling nvim-treesitter nvim-treesitter-textobjects resolves the issue. I wonder if you could conditionally disable tree-sitter depending on the file size and type?

**Reply by amaanq** [reply]
Not that I'm aware of, I'm hoping we'll have async injections to fix that end of the problem, I didn't know textobjects caused the lag too

**Reply by hakonharnes** [reply]
I don't know if textobjects cause lag, but I disabled it when disabling treesitter, because I'm assuming it won't work without it?

**Comment by gengdz** [comment]
It works if I disable nvim-treesitter! 
now the question becomes **how do I disable nvim-treesitter when the file size is over 1MB.**

I know the general structure of the code, but I don't know how to disable this plugin

```lua
vim.api.nvim_create_autocmd("BufReadPost", {
  pattern = { "*" },
  callback = function()
    local filesize = vim.fn.getfsize(vim.fn.expand("%:p"))
    if filesize < 1024 * 1024 then
      return
    end

  // TODO disable nvim-treesitter
  end,
})

```

**Reply by folke** [reply]
`vim.treesitter.stop()`
